### PR TITLE
docs: improve Factories

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
improve the description for `Factories`.

Related to #9181 #8914

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/concepts/factories.rst [hunk 1]
```diff
@@ -334,14 +334,14 @@ Set the following property to ``true`` in **app/Config/Optimize.php**::
...
 
-Since v4.5.0, you can enable this with the ``spark optimize`` command.
+Or you can enable this with the ``spark optimize`` command.
 
 .. note::
     This property cannot be overridden by
     :ref:`environment variables <configuration-classes-and-environment-variables>`.
 
 .. note::
-    Prior to v4.5.0, uncomment the following code in **public/index.php**::
```
Comment (neznaika0) [inline user_guide_src/source/concepts/factories.rst:344]: Why? The cache was added in **4.5.0**. For **4.4.8**, you need to uncomment the lines

File: user_guide_src/source/concepts/factories.rst [hunk 2]
```diff
@@ -334,14 +334,14 @@ Set the following property to ``true`` in **app/Config/Optimize.php**::
 
     public bool $configCacheEnabled = true;
 
-Since v4.5.0, you can enable this with the ``spark optimize`` command.
+Or you can enable this with the ``spark optimize`` command.
```
Comment (kenjis) [inline user_guide_src/source/concepts/factories.rst:337]: I think it is better to keep "Since v4.5.0", because devs who use old CI4 version could know they cannot use the feature when reading the latest user guide.

File: user_guide_src/source/concepts/factories.rst [hunk 3]
```diff
@@ -334,14 +334,14 @@ Set the following property to ``true`` in **app/Config/Optimize.php**::
...
 
 .. note::
     This property cannot be overridden by
     :ref:`environment variables <configuration-classes-and-environment-variables>`.
 
 .. note::
-    Prior to v4.5.0, uncomment the following code in **public/index.php**::
+    Prior to v4.4.x, uncomment the following code in **public/index.php**::
```
Comment (kenjis) [inline user_guide_src/source/concepts/factories.rst:344]: Why this change? I think "Prior to v4.5.0" is correct.
It means before v4.5.0 (4.4.x and before) and v4.5.0 is not included.

File: user_guide_src/source/concepts/factories.rst [hunk 4]
```diff
@@ -274,7 +274,7 @@ Config Caching
 
 .. important:: Do not use this feature unless you have carefully read this section
     and understand how this feature works. Otherwise, your application will not
-    function properly.
+    function properly. Escpecially when you using function ``config()``, you must extends ``BaseConfig``.
```
[the remark this answers is not in the document]
Comment (neznaika0) [reply user_guide_src/source/concepts/factories.rst:277]: I agree.
There is no need to extend `BaseConfig`.
This includes replacing the `Registrar`, which is not necessary.
We won't be able to load `Autoload`, `Paths` - autoload doesn't work

File: user_guide_src/source/concepts/factories.rst (hunk 2 above)
Comment (ddevsr) [reply user_guide_src/source/concepts/factories.rst:337]: How to Enable Config Caching have info version added 4.5.0, I think that includes 1 segment
image

File: user_guide_src/source/concepts/factories.rst (hunk 3 above)
Comment (ddevsr) [reply user_guide_src/source/concepts/factories.rst:344]: Sorry my bad english. By the way, Config caching ready in version 4.4.x, i think better specify version `4.4.x`

File: user_guide_src/source/concepts/factories.rst (hunk 2 above)
Comment (kenjis) [reply user_guide_src/source/concepts/factories.rst:337]: Okay.

File: user_guide_src/source/concepts/factories.rst (hunk 3 above)
Comment (kenjis) [reply user_guide_src/source/concepts/factories.rst:344]: Ah, you mean "**In v4.4.x**", not "**Prior to v4.4.x**"?
Because Config caching was introduced in v4.4.0.

Comment (ddevsr) [reply user_guide_src/source/concepts/factories.rst:344]: Nah thats it

File: user_guide_src/source/concepts/factories.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (ddevsr) [reply user_guide_src/source/concepts/factories.rst:277]: Yes, calling `config()` is important. However, when the cache is enabled, calling config() should trigger the `__set_state()` function. Create these functions or just extend BaseConfig? this not clear

[the remark this answers is not in the document]
Comment (ddevsr) [reply user_guide_src/source/concepts/factories.rst:277]: By the way, this segment important note in Config Caching

[the remark this answers is not in the document]
Comment (neznaika0) [reply user_guide_src/source/concepts/factories.rst:277]: There is no need to expand files for default. 
In fact, they should be read-only, without dynamic modification.

For others, the option depends on the context in the project. it is impossible to accurately describe the solution
We can only clarify that caching works when using `config()`

[the remark this answers is not in the document]
Comment (kenjis) [reply user_guide_src/source/concepts/factories.rst:277]: > Create these functions or just extend BaseConfig? this not clear

You must do either. If you need features in `BaseConfig`, you need to extend `BaseConfig`.

[the remark this answers is not in the document]
Comment (ddevsr) [reply user_guide_src/source/concepts/factories.rst]: I remove that notes, `Prerequisite` sections can explain

**Review by neznaika0 — APPROVED** [review]

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@ddevsr Thank you!
